flake.nix

```diff
@@ -1,16 +1,43 @@
 {
   inputs = {
-    core.url = "github:auxolotl/packages";
+    core.url = "github:auxolotl/core";
 
     javascript = {
       url = "github:auxolotl/javascript";
-      inputs.top-level.follows = "";
+      inputs.core.follows = "core";
     };
   };
 
-  outputs = { self, core, javascript }: {
-    inherit core;
+  outputs =
+    {
+      self,
+      core,
+      javascript,
+    }:
+    let
+      forAllSystems =
+        function:
+        core.lib.genAttrs core.lib.systems.flakeExposed (
+          system:
+          function {
+            nixPackages = core.nixPackages.${system};
+            auxPackages = core.auxPackages.${system};
+          }
+        );
+    in
+    {
+      inherit (core) lib nixPackages;
 
-    javascript = javascript.topLevelOut { inherit core; };
-  };
+      auxPackages = forAllSystems (
```

Comment:
Is there a reason for us to not use `packages` or `legacyPackages`? `packages` could be a problem given how much could end up exported here, but `legacyPackages` shouldn't be an issue and would also line up with the existing flake schema.

Reply:
> packages could be a problem given how much could end up exported here

this was my reasoning

> but legacyPackages shouldn't be an issue and would also line up with the existing flake schema.

`auxPackages` is already used in core, and i think its a good idea to move away from the `legacyPackages` term as it's pretty confusing to new user

Reply:
Agreed, `legacyPackages` is a terrible name - however not using it does leave us with a flake that doesn't pass `nix flake check`. Not a huge deal, and honestly? Maybe it's a fair deviation. Not a super massive concern overall.

Reply:
> Agreed, legacyPackages is a terrible name - however not using it does leave us with a flake that doesn't pass nix flake check

![image](https://github.com/auxolotl/top-level/assets/48872998/301a629c-40b7-494a-8b6a-4b49df0e0166)
we do. unknown outputs are only a warning after all

for the future, this could be solved by flake schemas...but i think that would be more of a Lix conversation :p